packages/esm-commons-lib/src/components/patient-table/patient-table.component.tsx

```diff
@@ -0,0 +1,383 @@
+  value?: string | number;
+}
+
+interface PatientTableProps {
+  autoFocus?: boolean;
+  columns: Array<PatientTableColumn>;
+  patients: Array<any>;
+  isFetching?: boolean;
+  isLoading: boolean;
+  mutateListDetails: () => void;
+  mutateListMembers: () => void;
+  pagination: {
+    usePagination: boolean;
+    currentPage: number;
+    onChange(props: any): any;
+    pageSize: number;
+    totalItems: number;
+    pagesUnknown?: boolean;
+    lastPage?: boolean;
+  };
+  style?: CSSProperties;
+}
+
+interface PatientTableColumn {
+  key: string;
+  header: string;
+  getValue?(patient: any): any;
+  link?: {
+    getUrl(patient: any): string;
+  };
+}
+
+export const PatientTable: React.FC<PatientTableProps> = ({
+  columns,
+  isFetching,
+  isLoading,
+  mutateListDetails,
+  mutateListMembers,
+  pagination,
+  patients,
+}) => {
+  const { t } = useTranslation();
+  const id = useId();
+  const layout = useLayoutType();
+  const responsiveSize = isDesktop(layout) ? 'sm' : 'lg';
+
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [membershipUuid, setMembershipUuid] = useState('');
+  const [patientName, setPatientName] = useState('');
+  const [searchTerm, setSearchTerm] = useState('');
+  const [showConfirmationModal, setShowConfirmationModal] = useState(false);
+  const debouncedSearchTerm = useDebounce(searchTerm);
+
+  const filteredPatients = useMemo(() => {
+    if (!debouncedSearchTerm) {
+      return patients;
+    }
+
+    return debouncedSearchTerm
+      ? fuzzy
+          .filter(debouncedSearchTerm, patients, {
+            extract: (patient: any) => `${patient.name} ${patient.identifier} ${patient.sex}`,
+          })
+          .sort((r1, r2) => r1.score - r2.score)
+          .map((result) => result.original)
+      : patients;
+  }, [debouncedSearchTerm, patients]);
+
+  const tableRows: Array<typeof DataTableRow> = useMemo(
+    () =>
+      filteredPatients.map((patient, index) => {
+        const row = {
+          id: String(index),
+        };
+        columns.forEach((column) => {
+          const value = column.getValue?.(patient) || patient[column.key];
+          row[column.key] = column.link ? (
+            <ConfigurableLink className={styles.link} to={column.link.getUrl(patient)}>
+              {value}
+            </ConfigurableLink>
+          ) : (
+            value
+          );
+        });
+        return row;
+      }) ?? [],
+    [columns, filteredPatients],
+  );
+
+  // const handleRemovePatientFromList = useCallback(async () => {
+  //   setIsDeleting(true);
+
+  //   try {
+  //     await removePatientFromList(membershipUuid);
+  //     mutateListMembers();
+  //     mutateListDetails();
+
+  //     showToast({
+  //       critical: true,
+  //       kind: 'success',
+  //       description: t('listUpToDate', 'The list is now up to date'),
+  //       title: t('patientRemovedFromList', 'Patient removed from list'),
+  //     });
+  //   } catch (error) {
+  //     showToast({
```

Comment:
Hey @pirupius, do you intend to use this code at some point?

Reply:
Yes. We might need to revisit this action but didn't want to go into a rabbit hole with each functionality. There was an error most likely endpoint related.